### DISCUSSION TITLE
Dynamically adjusting external IP target for the mobile app

### DISCUSSION THREAD

**Discussion by buildmine10** [immich-app/immich#14985] [Feature Request]
To access a home Immich server from outside the home network requires the Immich server to be port forwarded, then the user must use the external ip when logging into the Immich app.

My suggestion is that the Immich app logs in using the internal ip when connected to the home network, query for the current external ip of the house, and then use that external ip when connecting on networks other than the home network. (Aside from the port forwarding) this could be implemented in a way that it invisible to the user and it would allow for server access from outside the house while mitigating the issues that a dynamic ip causes.

**Comment by zackpollard** [comment]
You should use "dynamic dns" for this, there are plenty of services that provide this for free.

Here's an example: https://www.duckdns.org/
Here's a docker container you could use to automatically update it with your home IP: https://hub.docker.com/r/linuxserver/duckdns
